server/homebrew.model.js

```diff
@@ -46,8 +46,6 @@ HomebrewSchema.statics.get = function(query, fields=null){
 				unzipped = zlib.inflateRawSync(brews[0].textBin);
 				brews[0].text = unzipped.toString();
 			}
-			if(!brews[0].renderer)
-				brews[0].renderer = 'legacy';
```

Comment:
Is removing this going to cause old legacy brews to mistakenly render using V3? (although in general I agree with moving all of the "brew cleanup/defaulting" to a central location in homebrew.api.js)

Reply:
Short answer is "yes". Anything that was created and not updated since before the `renderer` value was added will use the default value; as this was done before v3 was available, by definition they must be written in legacy and may/likely will break when rendered in v3.
We can potentially differentiate on the `createdAt` or `updatedAt` values, as anything earlier than v3 going live should default to legacy. Alternatively, we could leave the default renderer value on loading as `legacy`, as all recent brews should include the renderer tag with an explicit value.

Reply:
Right, given that the only case of a missing value should be before V3, let's leave the default as "legacy".

Reply:
In that sense we use the term "default" in the sense of "values were missing, so we default back to what we guess they should be" rather than "this is what a blank brew should look like"

Reply:
I'm fairly sure that if we alter the "on load" default to something like `_.defaults(brew, loadDefaults, DEFAULT_BREW);` then any properties specified in `loadDefaults` will be - by definition - unable to be absent at the point that the `DEFAULT_BREW` properties are applied.
So `loadDefaults` only needs to be a diff of brew properties for loading vs saving.

I'll try to experiment with it later today.

Reply:
I've experimented with it a bit today, and `_.defaults` only works with properties that are `undefined` - but sometimes `renderer` is returning an empty string instead.
While we can simply use `assignWith` and craft a customizer object that replaces any falsy value with the default... but that would not necessarily work properly with actual Boolean values. I've pushed a commit with a custom `assign` function that is most of the way there, I think.